src/test/java/org/prebid/server/settings/S3ApplicationSettingsTest.java

```diff
@@ -0,0 +1,383 @@
+package org.prebid.server.settings;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.prebid.server.VertxTest;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.execution.Timeout;
+import org.prebid.server.execution.TimeoutFactory;
+import org.prebid.server.settings.model.Account;
+import org.prebid.server.settings.model.AccountAuctionConfig;
+import org.prebid.server.settings.model.AccountPrivacyConfig;
+import org.prebid.server.settings.model.StoredDataResult;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.NoSuchKeyException;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+
+@RunWith(VertxUnitRunner.class)
+public class S3ApplicationSettingsTest extends VertxTest {
+
+    private static final String BUCKET = "bucket";
+    private static final String ACCOUNTS_DIR = "accounts";
+    private static final String STORED_IMPS_DIR = "stored-imps";
+    private static final String STORED_REQUESTS_DIR = "stored-requests";
+    private static final String STORED_RESPONSES_DIR = "stored-responses";
+    @Rule
+    public final MockitoRule mockitoRule = MockitoJUnit.rule();
+    private Timeout timeout;
+
+    @Mock
+    private S3AsyncClient s3AsyncClient;
+    private Vertx vertx;
+
+    private S3ApplicationSettings s3ApplicationSettings;
```

Comment:
please call it `target` since it's under testing to distinguish from other dependencies